### DISCUSSION TITLE
misorientation axis for all possible neighboring lattice point pairs

### DISCUSSION THREAD

**Discussion by Debo-rah** [mtex-toolbox/mtex#1370] [Ask Anything]
Hi everyone,

I need to calculate the in grain misorientation axis (IGMA), which is the determination of the misorientation axes for all possible neighboring lattice-point pairs in the grain.

I found various ways to calculate misorientation axes, however, it seems most ways to calculate them (ebsd.mis2mean or grod) calculate the misorientation to the mean grain orientation instead of towards neighbouring points. 

I would need to calculate the misorientation axis between neighbouring ebsd points. 

Is this done by the following?

```
odf = calcDensity(ebsd.orientations);
% the uncorrelated
mdf = calcMDF(odf);
figure
plotAxisDistribution(mdf)
```

I also found this discussion with two possible scripts to get the misorientation axis calculated just like the Kernel average misorientation: 

https://gist.github.com/kilir/f51275526f78ffc4783ffe1aff88a048#file-kax_demo-m 

I have some questions about them:

1. )  Do both of them do what I want? I do not fully understand the code. The frist one (kax_demo) works with my data. If it is what I need, I have an additional question. Is there a way to adjust the kernel size in such a way that it is not squared? I am actually (in addition to the normal squared kernel) interested in the points above and underneath each other and not so much on the neighbours. I tried to put j or i to zero and only run one of the loops in the code, however, for j=0 it gave the same result as with the loop and for i=0 it returned the same axis for all points of my ebsd map. So I think that was a wrong approach and since the j-loop didnt do anything I am not sure whether this code is what i need.

2.) the second one (KAMAX) gives error messages. I think the problem here is that I dont know how to call the function.

I tried
[kamangle,kamaxis]=kamax(ebsd);
and got:
 
"
Undefined function 'spatialDecomposition' for input arguments of type 'double'.

Error in kamax (line 32)
[~,~,I_FD] = spatialDecomposition([ebsd.prop.x(:), ebsd.prop.y(:)],ebsd.unitCell,'unitCell');
"

However, the same line exists in the KAM code, which works when I call ebsd.KAM. 

If anyone understands the codes or knows another function of MTEX that calculates the misorientation axis between neighbouring points for a whole map, I would be very thankful!

Thanks for helping
Deborah

**Comment by kilir** [comment]
Hi,
no, what you found in the gist was an attempt to have an axis analogue to the KAM. 
However, if you want it for each individual neighboring point, you need to make a choice for which one you like to compute the misorientation.
Here's an example for one point pair (obviously there might be e.g. 8 possibilities)

```
mtexdata sharp

ebsd = ebsd.gridify;
% indices relative to center are ([8 2 7; 4 center 3; 6 1 5])
ind = squareNeighbors2(size(ebsd));

% example: misorientation axis relative to SE
% find indexed points
 p = 5
% make sure both points are indiexed;
p_ok = ebsd(ind(:,:,p)).isIndexed & ebsd.isIndexed;
e1 = ebsd(ind(:,:,p)); e1=e1(p_ok)
e2 = ebsd(p_ok);
mori = inv(e2.orientations).* e1.orientations;
ck = HSVDirectionKey(mori.axis)
figure
plot(e2,ck.direction2color(mori.axis))
```
Cheers,
Rüdiger

**Reply by Debo-rah** [reply]
Hi Rüdiger,

thanks  a  lot for your help. I have one more question to the code above. I have hexagonal gridded ebsd data. So each point only has 6 neighbours.
Do I then have to use 
`ind= hexNeighbors(size(ebsd));`
instead of  
`ind = squareNeighbors2(size(ebsd));`

And if so, how are the indices here relative to the center? Is it going clockwise? An which one is neighbour one?

If I want to just plot the misorientation axes of all neighbouring points into the fundamental region, can I do it like this then? 

```
% indexes of neighbours in hexagonal grid
ind= hexNeighbors(size(ebsd));
%preallocate misorientation array
mori=[];
% loop through all neighbours
for p=1:6
    % make sure both points are indiexed;
    p_ok = ebsd(ind(:,:,p)).isIndexed & ebsd.isIndexed;
    e1 = ebsd(ind(:,:,p)); 
    e1=e1(p_ok);
    e2 = ebsd(p_ok);
    mori_current = inv(e2.orientations).* e1.orientations;
    mori=cat(1,mori, mori_current);
end
figure
plot(mori.axis,'fundamentalRegion')

```

In this case I would have taken each pair twice right? So looping from p=1:3 woudl be what I need?

Thanks a lot!
Deborah

**Reply by kilir** [reply]
Hi,
yes, `hexNeighbors` would be the one for a hex-grid. The first index is to positive x, increasing clockwise. 
```
mtexdata titanium

ebsd = ebsd.gridify;
% indices of neighbors
ind = hexNeighbors(size(ebsd));

% indices relative to center are ([8 2 7; 4 center 3; 6 1 5])

p=50
b = 50
plot(ebsd(p,b),'faceColor','b')
hold on
col = parula(size(ind,3))
for j=1:size(ind,3)
    plot(ebsd(ind(p,b,j)),'faceColor',col(j,:))
    text(ebsd(ind(p,b,j)).x, ebsd(ind(p,b,j)).y, num2str(j))
end
hold off
```
Cheers,
Rüdiger
